strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/support/SpringSecurityLdapInternalsSupplier.java

```diff
@@ -0,0 +1,123 @@
+package org.carlspring.strongbox.controllers.security.ldap.support;
+
+import org.carlspring.strongbox.authentication.api.impl.ldap.LdapAuthenticator;
+import org.carlspring.strongbox.authentication.registry.AuthenticatorsRegistry;
+import org.carlspring.strongbox.authentication.support.AuthoritiesExternalToInternalMapper;
+import org.carlspring.strongbox.controllers.security.ldap.support.LdapConstants.LdapMessages;
+
+import javax.inject.Inject;
+import java.lang.reflect.Field;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.springframework.ldap.core.ContextSource;
+import org.springframework.security.ldap.authentication.AbstractLdapAuthenticator;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.ldap.search.FilterBasedLdapUserSearch;
+import org.springframework.security.ldap.search.LdapUserSearch;
+import org.springframework.security.ldap.userdetails.DefaultLdapAuthoritiesPopulator;
+import org.springframework.security.ldap.userdetails.LdapAuthoritiesPopulator;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ReflectionUtils;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class SpringSecurityLdapInternalsSupplier
```

Comment:
Cosmetic: Please, re-indent the chained methods below and make add an empty line before `return` statements.